### DISCUSSION TITLE
How to integrate with laravel lang files .

### DISCUSSION THREAD

**Discussion by KhairulAzmi21** [andrewdwallo/transmatic#17] [Q&A] [answered]
Hello andrewdwallo ,

I can see how this package works , but still I need to do it manually on laravel end in order to translate my system .

Right now, I dont know how to integrate this with our resource folder lang . How did you append new translation on each lang folder . 

**Do you mind share how would you use this in laravel project ?** 

I have look at the erpsaas you did implement this, but I still dont understand how it works .

**Comment by andrewdwallo** [comment]
Hey. Did you edit the file configuration setting in the configuration file for this package? You also need to make sure you use the `translate()` global helper or the `Transmatic::translate()` method.

**Comment by andrewdwallo** [comment]
Translations are appended in each language folder just based on the text you have passed into the method I mentioned above.

**Comment by andrewdwallo** [comment]
You also need to use json language files. Or you can use cache. And of course you need to make sure you have AWS Translate integrated properly.

**Reply by KhairulAzmi21** [reply]
I try use this in tinker . 

$translatedText = Transmatic::translate(text: 'Contacts', to: 'ms'); .

the translation working great, but how to append the translation to ms.json ?

**Reply by andrewdwallo** [reply, marked as answer]
> I try use this in tinker . 
> 
> 
> 
> $translatedText = Transmatic::translate(text: 'Contacts', to: 'ms'); .
> 
> 
> 
> the translation working great, but how to append the translation to ms.json ? 


Anytime you use the method in your application, such as for a label in the form, the package automatically appends the translation to your language file. If you want to generate them based on what you have in your default language file, you can take a look at the commands you can run in the documentation.

**Reply by KhairulAzmi21** [reply]
Thanks ! its working . 

I have question , can we translate what user enter in database ? for example I have roles . 

User will insert roles such as Finance, Admin etc . 

before call the data, i will translate first . do you think thats a good practice ?

**Reply by andrewdwallo** [reply]
Glad you got it to work! I don't really see the need to keep translated text in the database. The package handles translations already using lang files... but yeah you can store translated text if you want, just as long as you don't try to pass text that is already translated into the method..

**Reply by KhairulAzmi21** [reply]
Sorry, let me rephrase my question . 

for example I we have table category , user will insert category like Fashion,Gaming,Education etc .. this value is on database .

so how you approach to translate value from database ? 

before this I use to add new column on category like name_ms, name_ar, name_fr for each translation . but I find it hassle because each time I have new language , I will add another column ..

**Reply by andrewdwallo** [reply]
All you have to do is pass the text received from the database into the translate method. Then you can display it if you want to. As long as you aren't translating already translated text.

**Reply by KhairulAzmi21** [reply]
I have one last question , how you handle to check the text is already translated or not ?

**Reply by andrewdwallo** [reply]
The package doesn't handle this by default and would require a more sophisticated setup. You could try maybe storing the language of the text passed into the database compared to the current language of your app.. or if your using the "to" parameter.

**Reply by andrewdwallo** [reply]
I'm not really sure any packages really handle that kind of thing..

**Reply by KhairulAzmi21** [reply]
Thank you for answering my question ! you are great man 😄

**Reply by andrewdwallo** [reply]
@KhairulAzmi21 Your welcome, enjoy!
